### PR TITLE
[doc] Fix documented mstatus reset value

### LINKED ISSUE #2054: Unexpected PoR value of mstatus
### Bug Description

The [documented](https://ibex-core.readthedocs.io/en/latest/03_reference/cs_registers.html#machine-status-mstatus) power-on-reset value of the `mstatus` CSR (0x300) is 0x0000_1800.  The actual PoR value is 0x0000_0080.

I believe that the documented PoR value is the proper one, but the coded PoR has been in the Ibex code for three years so this is as much a question as it is a bug report.

## Observed Behavior

In simulations of our fork of Ibex, coming out of reset, the read value of `mstatus` is 0x0000_0080.  This is matches the coded PoR in the [RTL](https://github.com/lowRISC/ibex/blob/a7845832a2895ac3be8e28f41b87992207daee30/rtl/ibex_cs_registers.sv#L859):
```
localparam status_t MSTATUS_RST_VAL = '{mie:  1'b0,
                                          mpie: 1'b1,
                                          mpp:  PRIV_LVL_U,
                                          mprv: 1'b0,
                                          tw:   1'b0};
```
Given that [PRIV_LVL_U](https://github.com/lowRISC/ibex/blob/a7845832a2895ac3be8e28f41b87992207daee30/rtl/ibex_pkg.sv#L208) is 2'b00, this yields a PoR of 0x0000_0080.

## Expected Behavior

Coming out of reset, the read value of `mstatus` should be 0x0000_1800.  The RTL definition should look like this:
```
localparam status_t MSTATUS_RST_VAL = '{mie:  1'b0,
                                          mpie: 1'b0,
                                          mpp:  PRIV_LVL_M,
                                          mprv: 1'b0,
                                          tw:   1'b0};
```

## Steps to reproduce the issue

I have a test-program in CORE-V-VERIF for the CVE2, but do not have an Ibex test for this.  Having said that, visual inspection easily supports what we are seeing in simulation.

## My Environment

The OpenHW Group has created a fork of the Ibex which we are branding as the [CVE2](https://github.com/openhwgroup/cve2/tree/main).  The first release of this project will be the CV32E20.

**EDA tool and version:**

Questasim 2021.4_4

**Operating system:**

CentOS 7.9.2009

**Version of the Ibex source code:**

The current head as of 2023-06-22 @ 10:35 EDT: `a784583`.

### PR DESCRIPTION
**Pull request by GregAC**
Fixes #2054

### CODE REVIEW

**Review by hcallahan-lowrisc — APPROVED** [review]

File: examples/sw/simple_system/common/crt0.S [hunk 1]
```diff
@@ -46,6 +46,8 @@ reset_handler:
   mv x30, x1
   mv x31, x1
 
+  csrr x1, mstatus
```
Comment (rswarbrick) [inline examples/sw/simple_system/common/crt0.S:49]: No big deal, but could you write something in the commit message that explains why this change is in this commit?

**Review by rswarbrick — APPROVED** [review]
Minor tidiness question, but this looks sensible otherwise!
